sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/online_deployment.py

```diff
@@ -314,6 +315,36 @@ def _merge_with(self, other: "OnlineDeployment") -> None:
             self.instance_count = other.instance_count or self.instance_count
             self.instance_type = other.instance_type or self.instance_type
 
+    def _filter_datastore_to_rest_object(self):
+        # temporarily storing the data collector in the properties since it is no part of the contract
+        # will be removed once the contract is fixed to reflect data collector attribute
+        if is_private_preview_enabled() and self.data_collector:
+            non_flat_data = {}
+            non_flat_data["data_collector"] = self.data_collector._to_dict()
+            flat_data = flatten(non_flat_data, ".")
+            flat_data_keys = flat_data.keys()
+            for k in flat_data_keys:
+                self.tags[k] = flat_data[k]
+    @classmethod
+    def _filter_datastore_from_rest_object(cls, entity: "OnlineDeployment", deployment: RestOnlineDeploymentDetails) -> "OnlineDeployment":
+        # Data collector is private preview. If Private Preview environment variable is not enable
+        # data collector will be removed from tags. Data Collector values will be stored in tags
+        # until data collector is added to the contract.
+        if not is_private_preview_enabled():
+            del_key = []
+            for k in entity.tags:
+                if k.startswith("data_collector"):
+                    del_key.append(k)
+            if len(del_key) > 0:
+                for k in del_key:
+                    del entity.tags[k]
+        else:
+            unflat_data = unflatten(entity.tags, ".")
+            if unflat_data.get("data_collector", None):
```

Comment:
Do we officially support 3.8 as minimum yet?
If so you can do:
```
if val := unflat_data.get("data_collector", None):
  entity.data_collector = val
```
This allows us to not have to index twice into the dict

Reply:
The lowest supported version is 3.7 as of now. Are there any suggested changes based on that information?

Reply:
The walrus syntax `:=` is only supported from 3.8+. So NTD here if we need to support 3.7.
Theoretically you can do:
```
value = unflat_data.get("data_collector", None)
if value: # we avoid indexing twice
  # code
  pass
```
But it's a nit. We don't need to optimize so much.